File: backend/app/Models/Trufi.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Trufi extends Model
{
    protected $table = 'trufis';
    protected $primaryKey = 'idtrufi';

    // Si Tu PK No Se Llama "id", Esto Ayuda A Eloquent
    public $incrementing = true;
    protected $keyType = 'int';

    protected $fillable = [
        'nom_linea',
        'costo',
        'frecuencia',
        'tipo',
        'descripcion',
        'estado',
        'sindicato_id',
    ];

    protected $casts = [
        'estado' => 'boolean',
        'costo'  => 'decimal:2',
    ];

    // Relaciones
    public function rutas()
    {
        return $this->hasMany(\App\Models\Trufiruta::class, 'idtrufi', 'idtrufi');
    }

    public function sindicato()
    {
        return $this->belongsTo(\App\Models\Sindicato::class, 'sindicato_id', 'id');
    }

    // Opcional: Compatibilidad Si En Algún Lado Aún Usas "nombre"
    public function getNombreAttribute()
    {
        return $this->nom_linea;
    }

    // Opcional: Compatibilidad Si En Algún Lado Aún Usas "nombre_sindicato"
    public function getNombreSindicatoAttribute()
    {
        return $this->sindicato?->nombre;
    }
}

File: fronted/lib/models/trufi.dart
// lib/models/trufi.dart
class Trufi {
  final int idtrufi;
  final String nomLinea;
  final double costo;
  final int frecuencia;
  final String? descripcion;
  final bool estado;
  final int? sindicatoId;
  
  Trufi({
    required this.idtrufi,
    required this.nomLinea,
    required this.costo,
    required this.frecuencia,
    this.descripcion,
    required this.estado,
    this.sindicatoId,
  });
  
  factory Trufi.fromJson(Map<String, dynamic> json) {
    return Trufi(
      idtrufi: json['idtrufi'],
      nomLinea: json['nom_linea'],
      costo: double.parse(json['costo'].toString()),
      frecuencia: json['frecuencia'],
      descripcion: json['descripcion'],
      estado: json['estado'] == 1,
      sindicatoId: json['sindicato_id'],
    );
  }
}
